gi-gtk-declarative-app-simple/src/GI/Gtk/Declarative/App/Simple.hs

```diff
@@ -167,24 +168,15 @@ assertRuntimeSupportsBoundThreads = unless rtsSupportsBoundThreads $ do
                      \flag)."
   exitFailure
 
-
 publishEvent :: Chan event -> event -> IO ()
 publishEvent mvar = void . writeChan mvar
 
-mergeProducers :: [Producer a IO ()] -> Producer a IO ()
-mergeProducers producers = do
-  (output, input) <- liftIO $ spawn unbounded
-  _               <- liftIO $ Async.mapConcurrently_ (fork output) producers
-  fromInput input
```

Comment:
`fromInput` didn't get evaluated until `mapConcurrently_` had returned - so none of the events from the inputs got delivered until all of the events had been produced!